graph/builder.go

```diff
@@ -1229,23 +1197,7 @@ func (b *Builder) addNode(node *models.LightningNode,
 func (b *Builder) AddEdge(edge *models.ChannelEdgeInfo,
 	op ...batch.SchedulerOption) error {
 
-	rMsg := &routingMsg{
-		msg: edge,
-		op:  op,
-		err: make(chan error, 1),
-	}
-
-	select {
-	case b.networkUpdates <- rMsg:
-		select {
-		case err := <-rMsg.err:
-			return err
-		case <-b.quit:
-			return ErrGraphBuilderShuttingDown
-		}
-	case <-b.quit:
-		return ErrGraphBuilderShuttingDown
-	}
+	return b.handleNetworkUpdate(edge, op...)
```

Comment:
Very nice decomposition! This gets me to think, what's the point of creating a `handleNetworkUpdate`, when it takes a `msg any`, do a type switch there, and call the corresponding `addNode/addEdge/updateEdge`, when we already know what we want here? I think we can directly call `addEdge` inside `AddEdge` etc. Then `handleNetworkUpdate` can just be `handleTopologyChange`, wdyt?

Reply:
yeah mostly agree - the reason for this was just so we dont have to repeat the code that is constant across all 3 calls. namely the error handling (see the `if IsError(err, ErrIgnored, ErrOutdated) {` check above) and then the topology notifications. 

So can remove this but then we are repeating that code in each handler.

Reply:
(see the other comment btw)